src/shared/bg_public.h

```diff
@@ -1471,6 +1471,18 @@ void     AddFlag(int &flags, int flag);
 void     RemoveFlag(int &flags, int flag);
 void     ToggleFlag(int &flags, int flag);
 
+// true if first is lower than second argument
+// this function is like ` strcmp(first, second)` except constexpr.
+// outside of compile-time evaluations, prefer strcmp.
+constexpr int constexpr_strcmp(const char *first, const char *second)
+{
+       return *first == *second
+               ? (*first == '\0'
+                       ? 0 // strings are equal
+                       : constexpr_strcmp(first+1, second+1) ) // we aren't done y
+               : (*first < *second ? -1 : 1); // these two chars are different, so
```

Comment:
I don't suppose we'll ever use this on non-ASCII characters, but for portability and consistency with strcmp, the characters should be converted to the 0-255 range before comparison (char may be signed). E.g. `(*first & 0xFF) - (*second & 0xFF)`

Reply:
Fixed